### PR TITLE
Test against more versions of Ruby

### PR DESCRIPTION
**Pull request by manhhung741**
I'd like to be able to use Parlour with more versions of Ruby. This change will run spec against different version of Ruby. It might fail on ruby v2.3 and v2.4. We'll fix it before merging this change.

### CODE REVIEW

**Comment by manhhung741** [comment]
Currently this is failing because of the `yield_self` calls. I think they are for supporting this kind of use case
```
      a.create_class(name: 'Foo') do |foo|
        foo.create_method(name: 'add_two_integers', parameters: [
          Parlour::RbiGenerator::Parameter.new(name: 'a', type: 'Integer'),
          Parlour::RbiGenerator::Parameter.new(name: 'b', type: 'Integer')
        ], return_type: 'Integer')
      end
```
I'm planning to change it to `tap`

**Comment by connorshea** [comment]
Couldn't we just use `yield(self)` to support older ruby versions?

**Comment by manhhung741** [comment]
We can use `yield self`. `tap` does the same thing. I just manually fixed the code so it didn't work. Will try to do it properly later

**Comment by connorshea** [comment]
I took the backport for `yield_self` from https://github.com/marcandre/backports/blob/a1255e938cff17bb1a27e7b77df0860700267fae/lib/backports/2.5.0/kernel/yield_self.rb to see if it'd work :)

Commit here: https://github.com/connorshea/parlour/commit/42761fdfa16d40eb157c70d2e5673358b700810a

**Comment by connorshea** [comment]
Next we need to change prepend to unshift https://blog.bigbinary.com/2017/12/19/array-prepend-and-array-append-in-ruby-2-5.html

File: lib/parlour/plugin.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def self.inherited(new_plugin)
       registered_plugins[T.must(new_plugin.name)] = new_plugin
     end
 
-    sig { params(plugins: T::Array[Plugin], generator: RbiGenerator).void }
+    sig { params(plugins: T::Array[Plugin], generator: RbiGenerator, allow_failure: T::Boolean).void }
```
Comment (manhhung741) [inline lib/parlour/plugin.rb]: @AaronC81 we probably should set up git push hook to run `srb tc` and disallow any error committed. 
Is this file tested? If this file is tested, why doesn't this cause a sorbet run time error :-?

**Comment by manhhung741** [comment]
Ah, it passes all the tests now, except `ruby-head` (which is allowed to fail). Thanks @connorshea!

**Comment by connorshea** [comment]
You might want to rebase this since master changed some of these lines to `yield_self(&block) if block`

**Comment by manhhung741** [comment]
It says no conflict when rebase :-?

**Comment by connorshea** [comment]
🤷‍♂ https://github.com/AaronC81/parlour/commit/75aedc7ae11831c631df73482175851321260cb1

**Comment by connorshea** [comment]
Oh I guess it just got rid of the part of the commit that changed it back to yield_self

**Comment by manhhung741** [comment]
@AaronC81 can you merge this diff & cut a new release (0.6.1)? I have some PR on sorbet-rails dependent on this change

**Comment by AaronC81** [comment]
Sorry for the delay, merged. I'll push the new version to RubyGems when I'm next at my laptop in a few hours.

**Comment by AaronC81** [comment]
@manhhung741 - I've pushed 0.6.1 of Parlour with these changes.

**Comment by manhhung741** [comment]
@AaronC81 thank you! I appreciate it

**Pull request merged**